crates/http/src/routes.rs

```diff
@@ -291,13 +278,45 @@ impl RouteMatch {
     }
 
     /// The named wildcards captured from the path, if any
-    pub fn named_wildcards(&self) -> &HashMap<String, String> {
-        &self.named_wildcards
+    pub fn named_wildcards(&self) -> HashMap<String, String> {
+        let Some((_, best_match)) = &self.best_match else {
+            return HashMap::new();
+        };
+        best_match
+            .captures()
+            .iter()
+            .map(|(k, v)| (k.to_owned(), v.to_owned()))
```

Comment:
Is it worth exploring the possibility of optimising away these allocations?  I guess they are going to get allocated anyway when we populate the headers though...